agent/tests/test_cmd.py

```diff
@@ -256,3 +258,487 @@ async def testRightEnactOrderForMultipleEndpointsAtOnce(self, time, time_is_read
+    fr = fake_request(
+        {
+            "v2/releases": {
+                "releases": [
+                    {
+                        "name": "Firefox-64.0-build1",
+                        "product": None,
+                        "data_version": None,
+                        "read_only": None,
+                        "rule_info": {},
+                        "scheduled_changes": [
+                            {
+                                "name": "Firefox-64.0-build1",
+                                "product": "Firefox",
+                                "data_version": 1,
+                                "read_only": False,
+                                "sc_id": 1,
+                                "scheduled_by": "bob",
+                                "when": 2222222222000,
+                                "change_type": "insert",
+                                "sc_data_version": 1,
+                                "complete": False,
+                                "signoffs": {},
+                            },
+                            {
+                                "name": "Firefox-64.0-build1",
+                                "path": ".platforms.Darwin_x86_64-gcc3-u-i386-x86_64.locales.en-US",
+                                "data_version": 1,
+                                "sc_id": 1,
+                                "scheduled_by": "bob",
+                                "when": 2222222222000,
+                                "change_type": "insert",
+                                "sc_data_version": 1,
+                                "complete": False,
+                                "signoffs": {},
+                            },
+                            {
+                                "name": "Firefox-64.0-build1",
+                                "path": ".platforms.Linux_x86-gcc3.locales.en-US",
+                                "data_version": 1,
+                                "sc_id": 2,
+                                "scheduled_by": "bob",
+                                "when": 2222222222000,
+                                "change_type": "insert",
+                                "sc_data_version": 1,
+                                "complete": False,
+                                "signoffs": {},
+                            },
+                        ],
+                        "product_required_signoffs": {"releng": 1},
+                        "required_signoffs": {"releng": 1},
+                    }
+                ]
+            }
+        }
+    )
+    monkeypatch.setattr(balrogagent.cmd.client, "request", fr)
+    time_is_ready = MagicMock(return_value=True)
+    monkeypatch.setattr(balrogagent.cmd, "time_is_ready", time_is_ready)
+    await balrogagent.cmd.run_agent(asyncio.get_event_loop(), "http://balrog.fake", "telemetry", auth0_secrets={}, once=True, raise_exceptions=True)
+
+    # Once per scheduled change
+    assert time_is_ready.call_count == 3
```

Comment:
Worth checking the call count for verify_signoff here ?

One case you don't cover here is signoff being required and is present.